x-pack/plugins/security_solution/server/lib/detection_engine/routes/index/create_preview_index_route.ts

```diff
@@ -70,10 +71,13 @@ export const createPreviewIndex = async (
     await setPolicy(esClient, index, previewPolicy);
   }
 
+  const ruleDataService = context.securitySolution.getRuleDataService();
+  const aadIndexAliasName = ruleDataService.getResourceName(`security.alerts-${spaceId}`);
+
   if (await templateNeedsUpdate({ alias: index, esClient })) {
     await esClient.indices.putIndexTemplate({
       name: index,
-      body: getSignalsTemplate(index) as Record<string, unknown>,
+      body: getSignalsTemplate(index, aadIndexAliasName) as Record<string, unknown>,
```

Comment:
I haven't been following the latest `Rule Preview` efforts closely so just curious here, but why are we setting up the preview index with aliases to the AAD index? Looks like it's getting setup with the signals' `fieldAliases` as well (within `getSignalsTemplate`). Is this for querying compatibility with existing components?

Reply:
We don't need the index alias, I just wanted to avoid having conditional logic in `getSignalsTemplate` and this route is going away entirely in https://github.com/elastic/kibana/pull/116374 (it never shipped in 7.16 so we can remove this route without issue).

After https://github.com/elastic/kibana/pull/116374, the preview index/template logic will be handled by the RuleDataService.

The field aliases do matter though, as attempting to write to a field that is defined as an alias in the mapping will result in an error. This means for the new AAD indices since the `signal.*` fields are still defined (as aliases now) we'll still have conflicts if we try to copy user's source data into those fields.

Reply:
Ahh, perfect -- thanks for explaining! 🙂